src/account/BiconomySmartAccountV2.ts

```diff
@@ -1513,21 +1514,62 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
       userOp.maxFeePerGas = gasFeeValues?.maxFeePerGas as Hex
       userOp.maxPriorityFeePerGas = gasFeeValues?.maxPriorityFeePerGas as Hex
 
-      userOp = await this.getPaymasterUserOp(
-        userOp,
-        buildUseropDto.paymasterServiceData
-      )
-      return userOp
+      if(buildUseropDto.gasOffset){
+        userOp = await this.estimateUserOpGas(userOp)
+
+        const {verificationGasLimitIncrement, preVerificationGasIncrement, callGasLimitIncrement, maxFeePerGasIncrement, maxPriorityFeePerGasIncrement} = buildUseropDto.gasOffset;
```

Comment:
we should describe well how to provide these
increment compared to which base value?